test/tests_for_sdkmr/EE2Runjob_test.py

```diff
@@ -0,0 +1,268 @@
+
+
+def assert_jobs_equal(got_job: Job, expected_job: Job):
+    """
+    Checks that the two jobs are equivalent, except that the 'updated' fields are checked that
+    they're within 1 second of each other.
+    """
+    # I could not get assert got_job == expected_job to ever work, or even
+    # JobRequirements1 == JobRequirements2 to work when their fields were identical.
+    # Also accessing the instance dict via vars() or __dict__ only produced {'_cls': 'Job'} so
+    # apparently MongoEngine does something very odd with the dict.
+    # Hence we do this disgusting hack instead. Note it will need to be updated any time a
+    # job field is added.
+
+    if not hasattr(got_job, "id"):
+        assert not hasattr(expected_job, "id")
+    else:
+        assert got_job.id == expected_job.id
+
+    # The Job class fills the updated field with the output of time.time on instantiation
+    # so we can't do a straight equality
+    assert abs(got_job.updated - expected_job.updated) < 1
+
+    job_fields = [
+        "user",
+        "authstrat",
+        "wsid",
+        "status",
+        "queued",
+        "estimating",
+        "running",
+        "finished",
+        "errormsg",
+        "msg",
+        "error",
+        "terminated_code",
+        "error_code",
+        "scheduler_type",
+        "scheduler_id",
+        "scheduler_estimator_id",
+        "job_output",
+        "condor_job_ads",
+        "child_jobs",
+        "batch_job",
+    ]
+
+    _super_hacky_equals(got_job, expected_job, job_fields)
+
+    if not got_job.job_input:
+        assert expected_job.job_input is None
+    else:
+        job_input_fields = [
+            "wsid",
+            "method",
+            "requested_release",
+            "params",
+            "service_ver",
+            "app_id",
+            "source_ws_objects",
+            "parent_job_id",
+        ]
+        _super_hacky_equals(got_job.job_input, expected_job.job_input, job_input_fields)
+
+        requirements_fields = ["clientgroup", "cpu", "memory", "disk", "estimate"]
+        _super_hacky_equals(
+            got_job.job_input.requirements,
+            expected_job.job_input.requirements,
+            requirements_fields,
+        )
+        # this fails, which should be impossible given all the fields above pass
+        # assert got_job.job_input.requirements == expected_job.job_input.requirements
+
+        cell_info_fields = ["run_id", "token_id", "tag", "cell_id", "status"]
+        _super_hacky_equals(
+            got_job.job_input.narrative_cell_info,
+            expected_job.job_input.narrative_cell_info,
+            cell_info_fields,
+        )
+        # again, this fails, even though the fields are all ==
+        # assert got_job.job_input.narrative_cell_info == (
+        #     expected_job.job_input.narrative_cell_info)
+
+
+def _super_hacky_equals(obj1, obj2, fields):
+    for field in fields:
+        assert getattr(obj1, field) == getattr(obj2, field), field
```

Comment:
To aid future troubleshooting, I would either add in a comment to the effect of  "An `AttributeError` will be raised if the field is not present in the object" or add in a test that both objects have those attributes.

Reply:
Added the comment